sort/merge_sort.go

```diff
@@ -0,0 +1,34 @@
+package sort
+
+func mergeSort(arr []int) []int {
+	if len(arr) <= 1 {
+		return arr
+	}
+	mid := len(arr) / 2
+	left := make([]int, mid)
```

Comment:
### Runtime Error Handling

The `make` function used to create slices `left` and `right` does not handle negative lengths, which theoretically could happen if `mid` or `len(arr) - mid` were negative. While this is highly unlikely due to the nature of the algorithm, it's good practice to ensure these values are non-negative before creating slices:

```go
if mid < 0 || len(arr) - mid < 0 {
    panic("Invalid slice length")
}
```